src/toil/jobStores/aws/jobStore.py

```diff
@@ -430,28 +432,38 @@ def get_empty_file_store_id(self, jobStoreID=None, cleanup=False, basename=None)
         return info.fileID
 
     def _import_file(self, otherCls, uri, shared_file_name=None, hardlink=False, symlink=False):
-        if issubclass(otherCls, AWSJobStore):
-            srcObj = get_object_for_url(uri, existing=True)
-            size = srcObj.content_length
-            if shared_file_name is None:
-                info = self.FileInfo.create(srcObj.key)
-            else:
-                self._requireValidSharedFileName(shared_file_name)
-                jobStoreFileID = self._shared_file_id(shared_file_name)
-                info = self.FileInfo.loadOrCreate(jobStoreFileID=jobStoreFileID,
-                                                  ownerID=str(self.sharedFileOwnerID),
-                                                  encrypted=None)
-            info.copyFrom(srcObj)
-            info.save()
-            return FileID(info.fileID, size) if shared_file_name is None else None
-        else:
-            return super()._import_file(otherCls, uri, shared_file_name=shared_file_name)
+        try:
+            if issubclass(otherCls, AWSJobStore):
+                srcObj = get_object_for_url(uri, existing=True)
+                size = srcObj.content_length
+                if shared_file_name is None:
+                    info = self.FileInfo.create(srcObj.key)
+                else:
+                    self._requireValidSharedFileName(shared_file_name)
+                    jobStoreFileID = self._shared_file_id(shared_file_name)
+                    info = self.FileInfo.loadOrCreate(jobStoreFileID=jobStoreFileID,
+                                                      ownerID=str(self.sharedFileOwnerID),
+                                                      encrypted=None)
+                info.copyFrom(srcObj)
+                info.save()
+                return FileID(info.fileID, size) if shared_file_name is None else None
+        except ServerSideCopyProhibitedError:
+            # AWS refuses to do this copy for us
+            logger.warning("Falling back to copying via the local machine. This could get expensive!")
+            pass
```

Comment:
Also line 466, does `pass` do anything here?

Reply:
Not really; it's not needed to hold the block in existence because there's a logging line there.

I think I put it in to make it clear that I mean to discard this error, and so that if I decided not to have that logging line after all the code would still work.

Reply:
That makes sense.